### PR TITLE
Simplify handling of motif files

### LINKED ISSUE #2: Add more supported PWM formats
### Description of feature

Currently only the `PSEM` (Position-specific energy motif) format is supported, which is not widely used. With the following conversion steps we could accept the more popular `JASPAR` and `transfac` formats and convert them into the `PSEM` and `MEME` format as needed. The `MEME` format will be necessary for `SNEEP`

```[tasklist]
### Tasks
- [x] `JASPAR` to `transfac` using [this script](https://github.com/SchulzLab/TEPIC/blob/master/Code/JASPAR_to_PSCM.py) (optional, only if input is provided as `JASPAR`)
- [x] `transfac` to `PSEM` using [this script](https://github.com/SchulzLab/STARE/blob/main/Code/PSCM_to_PSEM.cpp)
- [x] `transfac` to `MEME` using [transfac2meme](https://meme-suite.org/meme/doc/transfac2meme.html) which should be available in [this container](https://biocontainers.pro/tools/meme)
- [ ] Use `MEME` output in `FIMO` subworkflow
```

### PR DESCRIPTION
**Pull request by nictru**
This PR adds a new subworkflow `MOTIFS` which handles the conversion of several input formats into the formats needed by the pipeline. If no motif file is provided to the pipeline, it can fetch motifs from `JASPAR` based on the `taxon_id`. It also performs the motif filtering that was part of the `PEAKS` subworkflow earlier.

The outputs of this subworkflow can also be used to simplify the `FIMO` subworkflow.

### CODE REVIEW

**Comment by nictru** [comment]
Requires #5 to be merged first

**Comment by LeonHafner** [comment]
I updated the `FIMO` workflow by removing the `DOWNLOAD_JASPAR` process.
The MEME file from `MOTIFS` is now used in the `FIMO/FILTER_MOTIFS` process, which previously used the output of `DOWNLOAD_JASPAR`.

Currently `FIMO` is still using the pwm file to convert gene symbols to jaspar IDs. Since this mapping should also be part of the MEME file from `MOTIFS`, we could possible also use this one for the mapping and remove the pwm completely from that subworkflow. What do you mean @nictru?

**Comment by nictru** [comment]
Yes, that's the way to go
Since the PWM file should now contain the same information as the motif file already used, we can safely remove it.

File: subworkflows/local/fimo.nf [hunk 1]
```diff
@@ -14,25 +13,18 @@ workflow FIMO {
...
     main:
         ch_versions = Channel.empty()
 
-        JASPAR_MAPPING(tf_ranking, pwm)
-
-        JASPAR_DOWNLOAD()
-
-        FILTER_MOTIFS(JASPAR_MAPPING.out.jaspar_ids, JASPAR_DOWNLOAD.out.motifs)
+        JASPAR_MAPPING(tf_ranking, motifs_meme)
+        FILTER_MOTIFS(JASPAR_MAPPING.out.jaspar_ids, motifs_meme)
```
Comment (nictru) [inline subworkflows/local/fimo.nf]: `JASPAR_MAPPING` and `FILTER_MOTIFS` aim to keep only the motifs for transcription factors found significant by the pipeline. Splitting this into two processes made sense in the original implementation with a dedicated download from JASPAR, but now this should be done in a single process.

**Pull request merged**